src/jit/emitxarch.cpp

```diff
@@ -2903,14 +2903,27 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
         varNum = tmpDsc->tdTempNum();
         offset = 0;
     }
+    else
+    {
+        // At this point we must have a memory operand that is a contained indir: if we do not, we should have handled
+        // this instruction above in the reg/imm or reg/reg case.
+        assert(mem != nullptr);
+        assert(memBase != nullptr);
+
+        if (memBase->OperGet() == GT_LCL_VAR_ADDR)
+        {
+            varNum = memBase->AsLclVarCommon()->GetLclNum();
+            offset = 0;
```

Comment:
This is a case of `GT_IND(GT_LCL_VAR_ADDR)` specifically, not `GT_IND(addr)`: `memBase` is set to `op1` of the indirection (not `Base()`). However, the asserts you suggest are reasonable future-proofing in case we ever decide to loosen the pattern matching, so I will add them.